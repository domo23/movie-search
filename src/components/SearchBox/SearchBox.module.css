.searchInput {
    width: 100%;
    padding: 5px 5px;
    display: inline-block;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
}

.main {
    display: flex;
    height: 100%;
    align-items: center;
}

.searchInput:focus {
    outline: none;
}