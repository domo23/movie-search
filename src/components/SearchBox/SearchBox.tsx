import React from "react";
import { useState } from "react";
import styles from "./SearchBox.module.css";
import { debounce } from "lodash";

export type Props = {
  onChange: (value: string) => void
}

function SearchBox({onChange}: Props) {
  const [formValue, setFormValue] = useState("");

  const handleChange = async function (
    event: React.ChangeEvent<HTMLInputElement>
  ) {
    const value = event.target.value;
    if (formValue === value) return;
    setFormValue(value);
    onChange(value);
  };

  return (
    <div className={styles.main} data-testid="search-box">
      <form>
        <input
          data-testid="search-box-input"
          className={styles.searchInput}
          placeholder="Search movie..."
          type="text"
          onChange={debounce(handleChange, 150)}
        />
      </form>
    </div>
  );
}

export default SearchBox;
