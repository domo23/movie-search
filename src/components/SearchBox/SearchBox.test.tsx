import React from "react";
import { fireEvent, render } from "@testing-library/react";
import SearchBox from './SearchBox';
import {Props} from './SearchBox';

function renderSearchForm(props: Partial<Props> = {}) {
    const defaultProps: Props = {
        onChange() {
        return;
      },
    };
    return render(<SearchBox {...defaultProps} {...props} />);
  }

it("sets movies based on input, snapshot test", () => {
    const { getByTestId } = renderSearchForm();
    const input = getByTestId('search-box');
    expect(input).toMatchSnapshot();
});

it("test handler function", async () => {
    jest.useRealTimers()
    const onChange = jest.fn();
    const { getByTestId } = renderSearchForm({onChange});
    const input = getByTestId('search-box-input');
   

    const event = {target: {value: "test"}};
    fireEvent.change(input, event);

    setTimeout(() => {
      expect(onChange).toHaveBeenCalledWith("test");
    }, 200)
});