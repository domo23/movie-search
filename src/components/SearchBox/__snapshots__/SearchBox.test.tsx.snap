// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets movies based on input 1`] = `
<div
  class="main"
  data-testid="search-box"
>
  <form>
    <input
      class="searchInput"
      data-testid="search-box-input"
      placeholder="Search movie..."
      type="text"
    />
  </form>
</div>
`;

exports[`sets movies based on input, snapshot test 1`] = `
<div
  class="main"
  data-testid="search-box"
>
  <form>
    <input
      class="searchInput"
      data-testid="search-box-input"
      placeholder="Search movie..."
      type="text"
    />
  </form>
</div>
`;
